### PR TITLE
TT-5315, TT-5319 fix hanging and white screen

### PR DESCRIPTION
**Pull request by gtryus**
- refactor react render
- memorize ref rendering
- memorize extras menu and action menu
- fix navigation warning errors
- move blue badge to right on section col
- upgrade to latest datasheet control
- fix bug snag errors

### CODE REVIEW

File: src/components/Workflow/PlanSheet.tsx [hunk 1]
```diff
@@ -740,6 +739,28 @@ export function PlanSheet(props: IProps) {
...
+                            flat={inlinePassages}
+                          />
+                        ) : (
+                          e
+                        ),
+                      readOnly:
+                        readonly ||
+                        passageTypeFromRef(e as string, inlinePassages) !==
+                          PassageTypeEnum.PASSAGE,
+                      className:
```
Comment (sarahentzel) [inline src/components/Workflow/PlanSheet.tsx:758]: Do we still need this special classname?

Comment (gtryus) [reply src/components/Workflow/PlanSheet.tsx:758]: I think if the user enters a bad reference, we show the error with this class name. It helps make sure we can post to Paratext.

Comment (gtryus) [reply src/components/Workflow/PlanSheet.tsx:758]: We could probably simplify the test since we know we are in the reference column.

Comment (gtryus) [reply src/components/Workflow/PlanSheet.tsx:758]: I simplified it.

**Pull request merged**